test/common_methods_invocations.py

```diff
@@ -827,6 +827,144 @@ def maybe_non_contig(tensor):
 
 
+def _compare_trilu_indices(
+        self, row, col, offset=0, dtype=torch.long, device='cpu'):
+    if row == 0 or col == 0:
+        # have to handle this separately as tril and triu does not take
+        # empty matrix as input
+        self.assertEqual(
+            torch.empty(0, 2, dtype=dtype, device=device).transpose(0, 1),
+            torch.tril_indices(row, col, offset, dtype=dtype, device=device))
+
+        self.assertEqual(
+            torch.empty(0, 2, dtype=dtype, device=device).transpose(0, 1),
+            torch.triu_indices(row, col, offset, dtype=dtype, device=device))
+
+    else:
+        self.assertEqual(
+            torch.ones(row, col, dtype=dtype, device='cpu')
+                 .tril(offset).nonzero().transpose(0, 1).to(device),
+            torch.tril_indices(row, col, offset, dtype=dtype, device=device))
+
+        self.assertEqual(
+            torch.ones(row, col, dtype=dtype, device='cpu')
+                 .tril(offset).nonzero().transpose(0, 1).to(device),
+            torch.tril_indices(row, col, offset, dtype=dtype, device=device))
+
+
+def _compare_large_trilu_indices(
+        self, row, col, offset=0, dtype=torch.long, device='cpu'):
+    l = torch.ones(row, col, dtype=dtype, device='cpu').tril(offset) \
+             .nonzero()[-100:-1, :].transpose(0, 1).to(device)
+    torch.cuda.empty_cache()
+
+    r = torch.tril_indices(
+        row, col, offset, dtype=dtype, device=device)[:, -100:-1]
+    self.assertEqual(l, r)
+    torch.cuda.empty_cache()
+
+    l = torch.ones(row, col, dtype=dtype, device='cpu').triu(offset) \
+             .nonzero()[-100:-1, :].transpose(0, 1).to(device)
+    torch.cuda.empty_cache()
+
+    r = torch.triu_indices(
+        row, col, offset, dtype=dtype, device=device)[:, -100:-1]
+    self.assertEqual(l, r)
+    torch.cuda.empty_cache()
+
+# (
+#   row
+#   col
+#   offset (optional)
+#   dtype (optional)
+# )
+tri_tests_args = [
+    (1, 1),
+    (3, 3),
+    (3, 3, 1),
+    (3, 3, 2),
+    (3, 3, 200),
+    (3, 3, -1),
+    (3, 3, -2),
+    (3, 3, -200),
+    (0, 3, 0),
+    (0, 3, 1),
+    (0, 3, -1),
+    (3, 0, 0),
+    (3, 0, 1),
+    (3, 0, -1),
+    (0, 0, 0),
+    (0, 0, 1),
+    (0, 0, -1),
+    (3, 6, 0),
+    (3, 6, 1),
+    (3, 6, 3),
+    (3, 6, 9),
+    (3, 6, -1),
+    (3, 6, -3),
+    (3, 6, -9),
+    (6, 3, 0),
+    (6, 3, 1),
+    (6, 3, 3),
+    (6, 3, 9),
+    (6, 3, -1),
+    (6, 3, -3),
+    (6, 3, -9),
+    (258, 253, 1, torch.float32),
+    (257, 258, 1, torch.float64),
+    (258, 258, 1, torch.short),
+    (3, 513, 1, torch.long),
+    (513, 3, 1, torch.int),
+    (513, 0, 1, torch.double),
+    (1024, 1024),
+    (1024, 1024, 500, torch.float32),
+    (1024, 1024, 1023),
+    (1024, 1024, -500),
+    (1023, 1025),
+    (1025, 1023, 1022),
+    (1024, 1024, -500),
+    (3, 2028),
+    (3, 2028, 1),
+    (3, 2028, -1),
+    (2028, 3),
+    (2028, 1),
+    (2028, 1, -1),
+    (5000, 5000),
+    (10000, 10000)
+]
+
+tri_large_tests_args = [
```

Comment:
How long does this take? It might be better if we don't run these in our CI if it takes too long / is too resource intensive

Reply:
On the gpudevserver, it takes ~2 minutes in total.

Reply:
Actually, I realize it is good to have these to test resolve_root_int's fallback binary search path. Let's leave them then unless they take too long to run

Reply:
Sure, I will comment out all tests but one, and add descriptions in `TensorFactories.cu` to remind developers if they would like to make some changes.